packages/api-client-core/src/GadgetConnection.ts

```diff
@@ -368,60 +361,18 @@ export class GadgetConnection {
   }
 
   private async waitForOpenedConnection(options: GadgetSubscriptionClientOptions): Promise<SubscriptionClient> {
-    let subscriptionClient = this.newSubscriptionClient(options);
-    let unsubscribes: Function[] = []; // eslint-disable-line @typescript-eslint/ban-types
-
-    let attempts = options.connectionAttempts || DEFAULT_CONN_ATTEMPTS;
-    const globalTimeout = options.connectionGlobalTimeoutMs || DEFAULT_CONN_GLOBAL_TIMEOUT;
-
-    const clearListeners = () => {
-      unsubscribes.forEach((fn) => fn());
-      unsubscribes = [];
-    };
+    let subscriptionClient: SubscriptionClient;
+    let unsubscribe: () => void;
 
     return await new Promise<SubscriptionClient>((resolve, reject) => {
-      const timeout = setTimeout(() => {
-        this.disposeClient(subscriptionClient);
-        wrappedReject(new Error("Timeout opening websocket connection to Gadget API"));
-      }, globalTimeout);
-
-      const retryOnClose = (event: unknown) => {
-        let message = "unknown close event";
-
-        if (isCloseEvent(event)) {
-          if (event.code == CloseCode.ConnectionAcknowledgementTimeout && attempts > 0) {
-            attempts -= 1;
-            this.disposeClient(subscriptionClient);
-            subscriptionClient = this.newSubscriptionClient(options);
-            resetListeners();
-            return;
-          }
-          message = `CloseEvent: ${event.reason}`;
-        }
-
-        clearTimeout(timeout);
-        reject(new Error(message));
-      };
-
-      const wrappedReject = (err: any) => {
-        clearTimeout(timeout);
-        reject(err);
-      };
-
-      const wrappedResolve = () => {
-        clearTimeout(timeout);
-        resolve(subscriptionClient);
-      };
-
-      const resetListeners = () => {
-        clearListeners();
-        unsubscribes.push(subscriptionClient.on("connected", wrappedResolve));
-        unsubscribes.push(subscriptionClient.on("closed", retryOnClose));
-        unsubscribes.push(subscriptionClient.on("error", wrappedReject));
-      };
-
-      resetListeners();
-    }).finally(clearListeners);
+      subscriptionClient = this.newSubscriptionClient({
+        ...options,
+        lazy: false,
+        onNonLazyError: reject,
+      });
+
+      unsubscribe = subscriptionClient.on("connected", () => resolve(subscriptionClient));
+    }).finally(() => unsubscribe?.());
```

Comment:
Is this `.` supposed to be here? Should it not be `unsubscribe?()`?

Reply:
I think the dot is required if you use `?`

Reply:
Function calls still need the `.` -- it does look weird at first glance though 😅 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#optional_chaining_with_function_calls